examples/Python/ChatApp/app.py

```diff
@@ -0,0 +1,192 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""
+Azure OpenAI Chat Application using Azure App Configuration.
+This script demonstrates how to create a chat application that uses Azure App Configuration
+to manage settings and Azure OpenAI to power chat interactions.
+"""
+
+import os
+from typing import List, Dict, Any
+from azure.core.credentials import TokenCredential
+from azure.identity import DefaultAzureCredential, get_bearer_token_provider
+from azure.appconfiguration.provider import load, SettingSelector, WatchKey
+from openai import AzureOpenAI
+from models import AzureOpenAIConfiguration, Message, ModelConfiguration
+
+
+def main():
+    # Create a credential using DefaultAzureCredential
+    credential = DefaultAzureCredential()
+    refresher = None
+
+    # Get the App Configuration endpoint from environment variables
+    app_config_endpoint = os.environ.get("AZURE_APPCONFIGURATION_ENDPOINT")
+    if not app_config_endpoint:
+        raise ValueError(
+            "The environment variable 'AZURE_APPCONFIGURATION_ENDPOINT' is not set or is empty."
+        )
+
+    try:
+        azure_openai_config = None
+        azure_client = None
+
+        def on_refresh_success():
+            azure_openai_config, azure_client = _create_ai_client(provider)
+
+        chat_app_selector = SettingSelector(key_filter="ChatApp:*")
+
+        global provider
+        # Create the configuration provider with refresh settings
+        provider = load(
+            endpoint=app_config_endpoint,
+            selects=[chat_app_selector],
+            credential=credential,
+            keyvault_credential=credential,  # Use the same credential for Key Vault references
+            trim_prefixes=["ChatApp:"],
+            refresh_on=[WatchKey(key="ChatApp:ChatCompletion:Messages")],
+            on_refresh_success=on_refresh_success,
+        )
+
+        azure_openai_config, azure_client = _create_ai_client(provider, credential)
```

Comment:
Can we just call `on_refresh_success` to force the initialization?

BTW, I think we should update our provider code to call `refresh_on` after the initial load of configuration. Otherwise, everybody needs to do this.

Reply:
This would be a breaking change for our provider library, plus calling `on_refresh_success` after the initial load seems odd and unexpected. We could add an `on_success` or something.

Reply:
How breaking? The on_refresh_success is meant to be called whenever the config is updated, which includes the config update for the first time (the initial load of configuration). If an app can break because on_refresh_success is triggerred, thre is something terribly wrong with the app.